### PR TITLE
Fix: allow colons in TS interface property names

### PR DESCRIPTION
**Pull request by smasala**
Developed with @Xyaren

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (3.3.x), `4.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @nicokoenig (2018/09)

### Description of the PR

Fix latest version by adding single quotes to properties with colons.

Example:
```typescript
export interface MyInterface {
    'my:name:property': string;
}
```

Original issue: https://github.com/OpenAPITools/openapi-generator/issues/1080

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java [hunk 1]
```diff
@@ -184,6 +186,25 @@ public String toVarName(String name) {
...
+        name = toVarName(name);
+        
+        if (hasIllegalCharacters(name)) {
+            name = "\'" + name + "\'";
+        }
+
+        return name;
+    }
+
+	private boolean hasIllegalCharacters(String name) {
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java]: a description in which context the characters are "illegal" would be nice

**Comment by macjohnny** [comment]
@smasala you need to replace the tab with spaces in https://github.com/OpenAPITools/openapi-generator/pull/1152/files#diff-328108658e03bbcf6cc8b931ffa7a63eR201

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java [hunk 2]
```diff
@@ -190,17 +190,24 @@ public String toVarName(String name) {
...
         if (hasIllegalCharacters(name)) {
             name = "\'" + name + "\'";
         }
 
         return name;
     }
 
-    private boolean hasIllegalCharacters(String name) {
+    /**
+     * Checks whether illegal characters are present in the given param
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java]: what is an illegal character and why?

**Comment by smasala** [comment]
@macjohnny this PR is now complete

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java [hunk 3]
```diff
@@ -184,6 +186,33 @@ public String toVarName(String name) {
         return name;
     }
 
+    @Override()
```
[the remark this answers is not in the document]
Comment (smasala) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java]: @wing328 and then leaving the `toPropertyName` completely out?

**Comment by smasala** [comment]
Thanks @wing328 and @macjohnny

**Pull request merged**